packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -523,10 +535,29 @@ export class SwapQuoter {
             gasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
         }
         // get the relevant orders for the makerAsset
-        let prunedOrders = await this._getSignedOrdersAsync(makerAssetData, takerAssetData);
+        let orders = await this._getSignedOrdersAsync(makerAssetData, takerAssetData);
+        if (options.intentOnFilling && options.apiKey) {
+            if (!this.rfqtTakerApiKeyWhitelist.includes(options.apiKey)) {
+                throw new Error('API key not permissioned for RFQ-T');
+            }
+            if (!options.takerAddress || options.takerAddress === '0x0000000000000000000000000000000000000000') {
```

Comment:
There is a `NULL_ADDRESS` in `constants.ts`

Reply:
Addressed in https://github.com/0xProject/0x-monorepo/pull/2541/commits/1670b21123d52ce1e5cb1fd71d477f9d89d9e0c5